src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/ApplicationInsightsScope.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    internal class ApplicationInsightsScope
```

Comment:
I'm actually going to make it more generic to be like `PropertyBagScope` or something. It has nothing to do with App Insights when you look at what it does.